Fix image device size cap at 1504px

wackerow opened the pull request:
## Description

Next.js docs: https://nextjs.org/docs/pages/api-reference/components/image#devicesizes

The default `deviceSizes` includes device sizes beyond the max width of 1504px that we use, causing images (especially heroes) to load a larger image than needed. Overriding the default `deviceSizes: [640, 750, 828, 1080, 1200, 1920, 2048, 3840],` with `deviceSizes: [640, 750, 828, 1080, 1200, 1504],` brings the image sizes back down to what we're used to, if not better.

Example: Homepage hero -> Production 800 kB, Next.js dev preview 2.7 MB, locally with above changes 556 kB

This _does_ however reduce the quality of the image slightly... Our Next.js preview site currently has slightly sharper images than production, and the above change makes them look like they do in production again. I tried adding `quality={100}` to these images to see if that would help (since currently it's using a quality of 75), but it doesn't seem to change anything. The `type` seems to be what's doing it, as the above changes make them load as `webp` instead of `png` (which is parity with production)

## Changes

- Overrides default decideSizes for images, capping largest device size at 1504px (global website max width)
- Reduces image sizes which were coming in significantly larger than in previous Gatsby repo

wackerow committed:
[fix] image device size cap at 1504px

Overrides default decideSizes for images, capping largest device size at 1504px (global website max width). Fixes image sizes which were coming in significantly larger than in previous Gatsby repo.

nhsz commented:
Great job, I just tested adding `1920` and images look a bit sharper while maintaining the size way below production current ones, I think it should be added, eg: with `/learn` page hero below (-53% size reduction)

**FORK:** 361kb

<img width="1509" alt="Screen Shot 2024-01-04 at 00 39 33" src="https://github.com/ethereum/ethereum-org-fork/assets/948922/2f691024-7f02-4c71-9498-036588619c59">

**PROD:** 768kb

<img width="1507" alt="Screen Shot 2024-01-04 at 00 40 10" src="https://github.com/ethereum/ethereum-org-fork/assets/948922/42f50e03-f57e-4386-bd8d-abe98934da3f">

wackerow commented:
Awesome, thanks @nhsz, I like that plan... will update!

wackerow committed:
patch: incremental image quality improvement

Re-add 1920 and 2048 default deviceSizes to array, leaving out only the XL width of 3840.

nhsz reviewed (commented):
IMO we should go up to `1920` only, with `2048` images are already bigger vs production (999kb vs 768kb in prod) and makes an improvement without degrading performance

<img width="1534" alt="Screen Shot 2024-01-04 at 15 01 59" src="https://github.com/ethereum/ethereum-org-fork/assets/948922/64fb1030-ffb9-41ca-aacd-04b0b0969a8e">

wackerow committed:
fix: remove 2048 device size, limit image size

wackerow commented:
@nhsz Sounds good! Just patched, should be good to go

nhsz reviewed (approved):
Nice, LGTM!

corwintines commented:
Awesome!
